### PR TITLE
Contest - thread filters & contest cancelling 

### PR DESCRIPTION
**Pull request by masvelio**
## Link to Issue
Closes: #7302 
Closes: #7463 

## Description of Changes
- added confirmation modal when contest is about to be cancelled
- edited Select component to so that it can show headers and dividers in the list. Also prevented edition for items other than topics
- adjusted HeaderWithFilters.tsx 
  -  combined topics dropdown with contest => modified logic so that it can be selected either one or the other
  - combined stages with status => status is only visible when contest filter is selected 

## Test Plan
- it all with mocked data so hard to test it properly
- good thing this is last PR that is "untestable", from the next up it will be possible to test some flows e2e

## Deployment Plan

1. safe to deploy

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/Select/Select.tsx [hunk 1]
```diff
@@ -19,13 +21,26 @@ export type SelectProps = {
   onSelect?: (
     v:
       | string
-      | { id: string | number; value: any; label: string; iconLeft?: IconName },
+      | {
+          id?: string | number;
+          value?: any;
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/Select/Select.tsx:26]: CI issue

File: packages/commonwealth/client/scripts/views/components/Select/Select.tsx [hunk 2]
```diff
@@ -103,19 +136,20 @@ export const Select = ({
...
-                        />
-                      ),
-                    })}
+                    {...(canEditOption &&
+                      !option.type && {
+                        iconRight: (
+                          <CWIconButton
+                            iconName="write"
+                            iconSize="small"
+                            onClick={async (e) => {
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/Select/Select.tsx:145]: CI issue

**Review by mzparacha — APPROVED** [review]
LGTM 🚀. Lets fix CI issues

**Review by Israellund — APPROVED** [review]
LGTM

**Pull request merged**